server/src/main/java/io/github/drp08/studypal/database/Friends.kt

```diff
@@ -0,0 +1,44 @@
+package io.github.drp08.studypal.database
+
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.eq
+import org.jetbrains.exposed.sql.Table
+import org.jetbrains.exposed.sql.and
+import org.jetbrains.exposed.sql.deleteWhere
+import org.jetbrains.exposed.sql.insert
+import org.jetbrains.exposed.sql.selectAll
+import org.jetbrains.exposed.sql.transactions.transaction
+
+object Friends : Table() {
+    val name = varchar("name", 50).uniqueIndex()
+    val friendName = varchar("friend_name", 50).uniqueIndex()
+}
+
+fun getFriends(name: String): List<String> {
+    return transaction {
+        Friends.selectAll().where { Friends.name eq name }.map { it[Friends.friendName] }
+    }
+}
+
+fun addFriend(name: String, friendName: String): Boolean {
+    val transaction = transaction {
+        Friends.insert {
+            it[Friends.name] = name
+            it[Friends.friendName] = friendName
+        }
+
+        Friends.insert {
+            it[Friends.name] = friendName
+            it[Friends.friendName] = name
+        }
+    }
+
+    return transaction.insertedCount == 1
```

Comment:
This is checking that only one row has been inserted, right? But we have two separate inserts, because A is a friend of B and vice versa, so why is it not "transaction.insertedCount == 2"?

Reply:
I'm not aware of this terminology. When I tested it "transaction.insertedColumns" returned 1 even though the test cases fully ran.

I suppose if you can test this if it actually inserts 2 rows by "sqlite3" into data.db. Or if you will, look through issues or make an issue on https://github.com/jetbrains/exposed about it.